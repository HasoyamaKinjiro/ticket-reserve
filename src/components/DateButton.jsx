import React, { forwardRef, useEffect, useState } from 'react';
import { Button } from '@mui/material';
import { useSelector } from 'react-redux';

import DateButtonStyles from '../styled/ChooseDayStyles/DateButtonStyles';

const DateButton = forwardRef(({ title, id, onClick, sx }, ref) => {
    const [isFocused, setIsFocused] = useState(false);
    const selectedDay = useSelector((state) => state.chooseDayState.selectedDay)

    useEffect(() => {
        if (id && id === selectedDay) {
            setIsFocused(true);
        } else {
            setIsFocused(false);
        }
    }, [id, selectedDay]);

    return (
        <Button
            id={id}
            color="secondary"
            variant="contained"
            sx={DateButtonStyles(isFocused, sx)}
            onClick={onClick}
            ref={ref}
        >
            {title}
        </Button>
    );
});

export default DateButton;
