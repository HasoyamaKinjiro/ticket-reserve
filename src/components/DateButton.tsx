import React, { forwardRef, MouseEventHandler, useEffect, useState } from 'react';
import { Button, SxProps, Theme } from '@mui/material';
import { useSelector } from 'react-redux';

import DateButtonStyles from '../styled/ChooseDayStyles/DateButtonStyles';

import { State } from '../redux/store';

interface DateButtonProps {
    title: string;
    id: string | undefined;
    onClick?: MouseEventHandler<HTMLButtonElement>;
    sx?: SxProps<Theme>;
}

const DateButton = forwardRef<HTMLButtonElement, DateButtonProps>(
    ({ title, id, onClick, sx }, ref) => {
    const [isFocused, setIsFocused] = useState<boolean>(false);
    const selectedDay = useSelector((state: State) => state.chooseDayState.selectedDay)

    useEffect(() => {
        if (id && id === selectedDay) {
            setIsFocused(true);
        } else {
            setIsFocused(false);
        }
    }, [id, selectedDay]);

    return (
        <Button
            id={id}
            color="secondary"
            variant="contained"
            sx={DateButtonStyles(isFocused, sx)}
            onClick={onClick}
            ref={ref}
        >
            {title}
        </Button>
    );
});

export default DateButton;
