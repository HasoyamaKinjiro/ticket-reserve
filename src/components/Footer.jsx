import React from 'react';
import { AppBar, Toolbar, Typography } from '@mui/material';
import { appBarFooter, toolbarFooter } from '../styled/FooterStyles';

const Footer = () => {
    return (
        <AppBar position="static" sx={appBarFooter}>
            <Toolbar sx={toolbarFooter}>
                <Typography variant="subtitle1">Â© 2024-2024 Ticket Reserve. All rights reserved.</Typography>
            </Toolbar>
        </AppBar>
    );
};

export default Footer;
