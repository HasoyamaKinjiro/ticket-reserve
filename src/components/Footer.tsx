import React from 'react';
import { AppBar, Toolbar, Typography } from '@mui/material';

import { appBarFooter, toolbarFooter } from '../styled/FooterStyles';

const Footer = () => (
    <AppBar position="static" sx={appBarFooter} role="contentinfo">
        <Toolbar sx={toolbarFooter}>
            <Typography variant="subtitle1">Â© 2024-2024 Ticket Reserve. All rights reserved.</Typography>
        </Toolbar>
    </AppBar>
);

export default Footer;
