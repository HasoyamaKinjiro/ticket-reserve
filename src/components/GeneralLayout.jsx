import React from 'react';
import { Header, Footer } from './index';
import { Outlet } from 'react-router-dom/dist';

const GeneralLayout = () => {
    return (
        <div>
            <Header/>
            <Outlet/>
            <Footer/>
        </div>
    );
};

export default GeneralLayout;
