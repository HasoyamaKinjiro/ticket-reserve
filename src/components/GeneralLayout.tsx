import React from 'react';
import { Outlet } from 'react-router-dom/dist';

import { Header, Footer } from './index';

const GeneralLayout = () => (
    <div>
        <Header/>
        <Outlet/>
        <Footer/>
    </div>
);

export default GeneralLayout;
