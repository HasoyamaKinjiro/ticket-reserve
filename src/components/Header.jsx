import React from "react";
import { AppBar, Toolbar, Typography, Icon } from "@mui/material";
import { LocalActivity } from "@mui/icons-material";

const Header = () => {
    return (
        <AppBar position="static" sx={{ mb: 3 }}>
            <Toolbar>
                <Typography variant="h6">Ticket reserve</Typography>
                <Icon color="inherit" sx={{ ml: 1 }}>
                    <LocalActivity />
                </Icon>
            </Toolbar>
        </AppBar>
    );
};

export default Header;
