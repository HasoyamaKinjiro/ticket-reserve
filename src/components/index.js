import Header from './Header';
import Footer from './Footer';
import GeneralLayout from './GeneralLayout';
import DateButton from './DateButton';
import OtherDays from './OtherDays';
import ChooseDay from '../containers/ChooseDay';
import PopUpPlaceButton from './PopUpPlaceButton';
import SessionCard from './SessionCard';
import SessionPopUp from './SessionPopUp';
import Sessions from '../containers/Sessions';

export {
    Header,
    Footer,
    GeneralLayout,
    DateButton,
    OtherDays,
    PopUpPlaceButton,
    SessionCard,
    SessionPopUp,
    ChooseDay,
    Sessions
};
