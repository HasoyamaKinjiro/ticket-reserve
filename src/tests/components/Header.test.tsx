import React from 'react';
import { render, screen } from '@testing-library/react';

import { Header } from '../../components';

describe('Header', () => {
    it('render with correct title', () => {
        render(<Header />);

        expect(screen.getByText('Ticket reserve')).toBeInTheDocument();
    });

    it('Header snapshot', () => {
        const header = render(<Header />);

        expect(header).toMatchSnapshot();
    });
});
