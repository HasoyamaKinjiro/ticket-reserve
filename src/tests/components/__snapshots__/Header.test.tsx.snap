// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-gdraet-MuiPaper-root-MuiAppBar-root"
        role="banner"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1wioqne-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Ticket reserve
          </h6>
          <span
            aria-hidden="true"
            class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-1i5ajve-MuiIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LocalActivityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12c0-1.1.9-2 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2m-4.42 4.8L12 14.5l-3.58 2.3 1.08-4.12-3.29-2.69 4.24-.25L12 5.8l1.54 3.95 4.24.25-3.29 2.69z"
              />
            </svg>
          </span>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-gdraet-MuiPaper-root-MuiAppBar-root"
      role="banner"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1wioqne-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Ticket reserve
        </h6>
        <span
          aria-hidden="true"
          class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-1i5ajve-MuiIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LocalActivityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12c0-1.1.9-2 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2m-4.42 4.8L12 14.5l-3.58 2.3 1.08-4.12-3.29-2.69 4.24-.25L12 5.8l1.54 3.95 4.24.25-3.29 2.69z"
            />
          </svg>
        </span>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
